src/MailHandling/SendNewPasswordMail.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\MailHandling;
+
+use App\Entity\User;
+use App\Entity\UserResetPasswordToken;
+use App\Repository\UserResetPasswordTokenRepository;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Symfony\Bridge\Twig\Mime\TemplatedEmail;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Mailer\Exception\TransportExceptionInterface;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Uid\Uuid;
+
+class SendNewPasswordMail implements LoggerAwareInterface
```

Comment:
I have 2 tables - user and user_reset_password_token. In controller I'm using $user and here $userToken. In this way in controller I only care about real user and here about a fictive userToken and about token which are only used for sending email. 
I prefer this way because creating token and userToken are only related with creating and sending email.... these 2 activities (creating token and userToken) are not related with anything else.

Reply:
See the message below